First try at dns zone

mykelalvis opened the pull request:
### Description

This PR add Azure DNS zone and record matching.

### Issues Resolved

Haven't added an issue yet

### Check List

- [ ] New functionality includes tests (Sort of)
- [ ] All tests pass
- [ ] `rake lint` passes (this didn't work for me before I made changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

mykelalvis committed:
First try at dns zone

mykelalvis committed:
First try at dns zone

mykelalvis committed:
First try at dns zone

mykelalvis committed:
First try at dns zone

mykelalvis committed:
Remove unnecessary var

mykelalvis committed:
Remove unnecessary var

mykelalvis committed:
Remove unnecessary var

mykelalvis committed:
Remove unnecessary var

mykelalvis committed:
More work on the dns_records

mykelalvis committed:
More work on the dns_records

mykelalvis committed:
More work on the dns_records

mykelalvis committed:
More work on the dns_records

mykelalvis committed:
Finish out the data mapping for dns records

mykelalvis committed:
Finish out the data mapping for dns records

mykelalvis committed:
Finish out the data mapping for dns records

mykelalvis committed:
Finish out the data mapping for dns records

mykelalvis committed:
Cleaner solution by moving to accessor methods in dns record object

mykelalvis committed:
Cleaner solution by moving to accessor methods in dns record object

mykelalvis committed:
Cleaner solution by moving to accessor methods in dns record object

mykelalvis committed:
Cleaner solution by moving to accessor methods in dns record object

mykelalvis committed:
Document update

mykelalvis committed:
Document update

mykelalvis committed:
Document update

mykelalvis committed:
Document update

mykelalvis committed:
Add some skeletal integration tests

Fix a bug in docs

mykelalvis committed:
Add some skeletal integration tests

Fix a bug in docs

mykelalvis committed:
Add some skeletal integration tests

Fix a bug in docs

mykelalvis committed:
Example update

mykelalvis committed:
Example update

mykelalvis committed:
Example update

mykelalvis committed:
Example update

mattray commented on libraries/azurerm_network_dns_records.rb:
```diff
@@ -0,0 +1,36 @@
+  def initialize(resource_group: nil, zone_name:, record_type: )
+    @record_type = record_type
+    dns_records = management.dns_records(resource_group: resource_group, zone_name: zone_name, record_type: record_type)
+#    puts "\n\ndns_records is \n#{dns_records}"
```
leftover debugging statement?

mattray commented on libraries/azurerm_network_dns_recordsets.rb:
```diff
@@ -0,0 +1,35 @@
+
+  def initialize(resource_group: nil, zone_name:    )
+    dns_recordsets = management.dns_recordsets(resource_group, zone_name)
+#    puts "Recordsets is \n#{dns_recordsets}"
```
leftover debugging statement?

mattray commented on libraries/azurerm_network_dns_zone.rb:
```diff
@@ -0,0 +1,48 @@
+    @name = name
+    @resource_group = resource_group
+    dns_zone = management.dns_zone(resource_group, name)
+#    puts "dns_zone is #{dns_zone}"
```
leftover debugging statement?

mattray commented on libraries/support/azure/rest.rb:
```diff
@@ -30,6 +30,7 @@ def get(path, params: {}, headers: {})
 
     def post(path, params: {}, headers: {}, body: nil)
       add_user_agent!(headers)
+      puts path
```
leftover debugging statement? This one's still active

mykelalvis committed:
Create AzureApisManagementService (and management utility)

mykelalvis committed:
Create AzureApisManagementService (and management utility)

mykelalvis committed:
Create AzureApisManagementService (and management utility)

mykelalvis committed:
Create AzureApisManagementService (and management utility)

mykelalvis committed:
Changes to make api management services less-terrible

mykelalvis committed:
Changes to make api management services less-terrible

mykelalvis committed:
Changes to make api management services less-terrible

mykelalvis committed:
Changes to make api management services less-terrible

gsreynolds committed:
gitignore .bundle and vendor

Signed-off-by: Gavin Reynolds <gavin@chef.io>

mykelalvis committed:
Debug/whitespace clean

mykelalvis committed:
Debug/whitespace clean

mykelalvis committed:
Debug/whitespace clean

mykelalvis committed:
Debug/whitespace clean

gsreynolds committed:
Add application gateway Terraform

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
For testing purposes just add a redirection config, otherwise appgw fails to provision because we have no real backend targets

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Add random string for app gw name

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Add TF outputs for appgw

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Add application_gateway and application_gateways resource

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Add integration tests for application gateway(s)

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Add test for application gateway ssl policy

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Add application gateway resource docs

Signed-off-by: Gavin Reynolds <gavin@chef.io>

gsreynolds committed:
Update resource links

Signed-off-by: Gavin Reynolds <gavin@chef.io>

Commit:
Bump version to 1.16.0 by Chef Expeditor

Obvious fix; these changes are the result of automation not creative thinking.

mykelalvis committed:
Adds for app service plan(s)

mykelalvis committed:
Adds for app service plan(s)

mykelalvis committed:
Adds for app service plan(s)

mykelalvis committed:
Updates to naming for app service plan

mykelalvis committed:
Updates to naming for app service plan

mykelalvis committed:
Updates to naming for app service plan

Commit:
Change file extensions from `.md.erb` to `.md`

- Update some typo in docs
- Rename `tag` to `tags` in azurerm_mysql_servers
- Update `data_disks` property of azurerm_virtual_machines to align with the documentation
- Update integration test

Signed-off-by: Omer Demirok <odemirok@chef.io>

Commit:
Fix an integration test failing the Build tests

This is due to not updating integration tests after this change:
https://github.com/inspec/inspec-azure/commit/e4cc8e1a9744f69cce83291abba6e4c0a2687e0e

Signed-off-by: Omer Demirok <odemirok@chef.io>

Commit:
Bump version to 1.16.1 by Chef Expeditor

Obvious fix; these changes are the result of automation not creative thinking.

Commit:
Remove left-over folder created at `rake lint` task

`rake lint` runs `inspec vendor` command before syntax check and this creates a `vendor` folder with the dependencies in it.
A second `rake lint` task detects the style issues in the `vendor` file and fails the `lint` task.
It should be removed after syntax check.

Signed-off-by: Omer Demirok <odemirok@chef.io>

rmoles committed:
Merge pull request #253 from inspec/remove_vendor

Remove left-over folder created at `rake lint` task

Commit:
Bump version to 1.16.2 by Chef Expeditor

Obvious fix; these changes are the result of automation not creative thinking.

mykelalvis committed:
Working on App Service PLan vnets

mykelalvis committed:
Working on App Service PLan vnets

mykelalvis committed:
Working on App Service PLan vnets

mykelalvis committed:
Merge branch 'feature/azurerm_dns' of github.com:infrastructurebuilder/inspec-azure into feature/azurerm_dns post-rebase

Commit:
Fix integration tests to reflect the last update

This relates to:
https://github.com/inspec/inspec-azure/pull/252/commits/d1db95f7f5d348638aa2b0e8f039eba282873ba5

Signed-off-by: Omer Demirok <odemirok@chef.io>

Commit:
Bump version to 1.16.3 by Chef Expeditor

Obvious fix; these changes are the result of automation not creative thinking.

dwizzle204 committed:
Add mariadb server resource

Signed-off-by: Dwayne Med <dwizzle204@gmail.com>

mykelalvis committed:
Merge from remote

Commit:
Merge pull request #256 from dwizzle204/master

Add mariadb server resource

Commit:
Bump version to 1.17.0 by Chef Expeditor

Obvious fix; these changes are the result of automation not creative thinking.

mykelalvis committed:
Merge branch 'develop' into feature/azurerm_dns

mykelalvis committed:
Modify README to more closely conform to portal
